app/ante/msg_ante.go

```diff
@@ -63,32 +67,33 @@ func (d MsgFilterDecorator) ValidateMsgs(ctx sdk.Context, msgs []sdk.Msg) error
 					return sdkerrors.Wrapf(sdkerrors.ErrInsufficientFunds, "insufficient initial deposit amount - required: %v", minInitialDeposit)
 				}
 			}
-		}
 
-		// deliver tx level msg filter
-		switch msg := msg.(type) {
-		// deprecated msgs
-		case *claimtypes.MsgClaim,
-			*farmingtypes.MsgCreateFixedAmountPlan,
-			*farmingtypes.MsgCreateRatioPlan,
-			*farmingtypes.MsgStake,
-			*farmingtypes.MsgUnstake,
-			*farmingtypes.MsgHarvest,
-			*farmingtypes.MsgRemovePlan,
-			*farmingtypes.MsgAdvanceEpoch:
-			return fmt.Errorf("%s is deprecated msg type", sdk.MsgTypeURL(msg))
+		// block mixed batch msg with regular msg
+		case *exchangetypes.MsgPlaceBatchLimitOrder,
+			*exchangetypes.MsgPlaceMMBatchLimitOrder,
+			*exchangetypes.MsgCancelOrder:
+			numMsg--
+			numBatchMsg++
+			if numMsg > 0 {
+				return fmt.Errorf("cannot mix batch msg and regular msg in one tx")
+			}
```

Comment:
```suggestion
			numMsg--
			numBatchMsg++
```

I think the check before the last `return` is enough.